From d8fdbbe185d075cee85d1fd4a673f656a03bf489 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:50:31 -0400
Subject: [tools] perf: Fix inverted error verification bug in thread__fork

Message-id: <1408953210-26343-144-git-send-email-jolsa@redhat.com>
Patchwork-id: 88838
O-Subject: [PATCH RHEL7.1 BZ1133083 143/322] perf tools: Fix inverted error verification bug in thread__fork
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 8d00be815c05ed0f0202f606bab4e54f98fd3b30
Author: David Ahern <dsahern@gmail.com>
Date: Tue Dec 10 21:35:38 2013 -0700

description
===========
Commit 1902efe7f for the new comm infra added the wrong check for return
code on thread__set_comm. err == 0 is normal, so don't return at that
point unless err != 0.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/thread.c b/tools/perf/util/thread.c
index 49eaf1d..e394861 100644
--- a/tools/perf/util/thread.c
+++ b/tools/perf/util/thread.c
@@ -126,7 +126,7 @@ int thread__fork(struct thread *thread, struct thread *parent, u64 timestamp)
   if (!comm)
    return -ENOMEM;
   err = thread__set_comm(thread, comm, timestamp);
-  if (!err)
+  if (err)
    return err;
   thread->comm_set = true;
  }
-- 
1.7.1